satpy/tests/test_composites.py

```diff
@@ -990,6 +989,18 @@ def load(self, arg):
         comp = StaticImageCompositor("name", filename="${TEST_IMAGE_PATH}/foo.tif", area='euro4')
         self.assertEqual(comp._cache_filename, "/path/to/image/foo.tif")
 
+        # URL and filename without absolute path
+        comp = StaticImageCompositor("name", url=remote_tif, filename="bar.tif")
+        self.assertEqual(comp._url, remote_tif)
+        self.assertEqual(comp._cache_filename, "bar.tif")
+
+        # No URL, filename without absolute path, use default data_dir from config
+        with mock.patch('os.path.exists') as exists:
+            exists.return_value = True
+            comp = StaticImageCompositor("name", filename="foo.tif")
+            self.assertEqual(comp._url, None)
+            self.assertEqual(comp._cache_filename, "/path/to/image/foo.tif")
```

Comment:
The tests are failing on Windows. I think this has to be `os.path.join('path', 'to', 'image', 'foo.tif')`, but I'm not sure how the root directory is handled.

Reply:
Thanks, Dave. I have fixed the test case, added an `os.path.sep` as root directory.